src/Clapi/Valuespace.hs

```diff
@@ -225,6 +225,9 @@ makeLenses ''Valuespace
 unvalidate :: Valuespace v -> Valuespace Unvalidated
 unvalidate (Valuespace tr ty x d u) = Valuespace tr ty x d u
 
+vsGetTree :: Valuespace v -> VsTree
+vsGetTree = _tree
```

Comment:
That `_tree` thing is underscored by convention because it was used by some template Haskell to generate some lenses on the `VsTree`. If you want to use it, you can either use it as you have here (in which case, I'm not sure what the alias buys us) or you can use it like a lens: `\vs -> view tree vs :: Valuespace v -> VsTree` (off the top of my head).

Reply:
(In fact, see `Show` instance below ;-) )